modules/transactionpool/transactionpool_test.go

```diff
@@ -655,3 +655,660 @@ func TestHeapFees(t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, graph := range graphs2 {
+		err := tpt.tpool.AcceptTransactionSet([]types.Transaction{graph[0]})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	_, err = tpt.miner.AddBlock()
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, graph := range graphs3 {
+		err := tpt.tpool.AcceptTransactionSet([]types.Transaction{graph[0]})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	_, err = tpt.miner.AddBlock()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Accept the first half of the first chunk of transactions in a random order.
+	firstIndices := fastrand.Perm(numGraphsPerChunk / 2)
+	for _, i := range firstIndices {
+		graph := graphs[i]
+		for _, txn := range graph[1:] {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{txn})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	// Accept the first half transactions of the second chunk in a random order.
+	mixSecond := fastrand.Perm(numGraphsPerChunk / 2)
+	for _, i := range mixSecond {
+		graph := graphs2[i]
+		for _, txn := range graph[1:] {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{txn})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	// Accept the rest of the first chunk in random order.
+	finishFirstChunk := fastrand.Perm(numGraphsPerChunk / 2)
+	for _, i := range finishFirstChunk {
+		graph := graphs[i+(numGraphsPerChunk/2)]
+		for _, txn := range graph[1:] {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{txn})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	// Accept the next fourth of the transactions of the second chunk in a random order.
+	moreSecond := fastrand.Perm(numGraphsPerChunk / 4)
+	for _, i := range moreSecond {
+		graph := graphs2[i+numGraphsPerChunk/2]
+		for _, txn := range graph[1:] {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{txn})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	// Mix in the first half transactions of the third chunk in a random order.
+	mixThird := fastrand.Perm(numGraphsPerChunk / 2)
+	for _, i := range mixThird {
+		graph := graphs3[i]
+		for _, txn := range graph[1:] {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{txn})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	// Accept the last remaining transactions of the second chunk in a random order.
+	finishSecond := fastrand.Perm(numGraphsPerChunk / 4)
+	for _, i := range finishSecond {
+		graph := graphs2[i+(numGraphsPerChunk/2)+(numGraphsPerChunk/4)]
+		for _, txn := range graph[1:] {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{txn})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	// Accept the remaining transactions from the last chunk.
+	finishThird := fastrand.Perm(numGraphsPerChunk / 2)
+	for _, i := range finishThird {
+		graph := graphs3[i+(numGraphsPerChunk/2)]
+		for _, txn := range graph[1:] {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{txn})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
```

Comment:
I'm thinking there's a better way to do this. Append all of the sets into a single large set, then do the first 250 as one `Perm`, do the second 500 as a second `Perm`, the third 500 as a third `Perm`, and the final 250 as a fourth `Perm`.

Since `Perm` I think only gives you 0-n, you'd have to do `perm+250` for the second set, `perm+750` for the third set, etc.

The result is a little different than what you have implemented above, but I think is overall a better mix.